### PR TITLE
applications: nrf_desktop: Add CPI control using buttons

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds CPI changing using two predefined buttons.

### CODE REVIEW

File: applications/nrf_desktop/src/modules/cpi_state.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	BUILD_ASSERT(MOTION_SENSOR_CPI_MAX > MOTION_SENSOR_CPI_MIN);
+
+	if ((state == STATE_OFF) || (sent_config_event != NULL)) {
+		return;
+	}
+
+	int update = step_cnt * CONFIG_DESKTOP_CPI_STATE_STEP_CHANGE;
+	int new_cpi = (int)cpi + update;
+
+	if (new_cpi < MOTION_SENSOR_CPI_MIN) {
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/cpi_state.c]: remember about integer promotion
`MOTION_SENSOR_CPI_MIN` should be assumed unsigned and so will require casting

File: applications/nrf_desktop/src/modules/cpi_state.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+			update_cpi();
+		}
+	}
+
+	return false;
+}
+
+static bool handle_config_event(const struct config_event *event)
+{
+	if (is_my_config_id(event->id)) {
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/cpi_state.c]: I am actually not in favor of this as we will have logic for parsing motion sensor configuration in two places...

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/cpi_state.c]: maybe we should keep entire logic related to this parsing here and pass a CPI value as an additional event to the sensor?

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/cpi_state.c]: Then we would have to assume that every mouse with configurable CPI would have this module enabled, which currently is not true.

File: applications/nrf_desktop/src/modules/cpi_state.c (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/cpi_state.c]: done

**Comment by MarekPieta** [comment]
There were big changes in config channel, we cannot use the module as it is. I will close this PR.

**Pull request closed without merging**